test/algorithms/ground_state_solvers/test_adapt_vqe.py

```diff
@@ -68,6 +70,61 @@ def test_default(self):
         res = calc.solve(self.problem)
         self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
 
+    def test_param_shift(self):
+        """test for param_shift method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="param_shift")
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_fin_diff(self):
+        """test for fin_diff method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="fin_diff", epsilon=1.0)
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_lin_comb(self):
+        """test for fin_diff method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="lin_comb", epsilon=1.0)
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_natural_gradients(self):
+        """test for fin_diff method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = NaturalGradient(
+            grad_method="lin_comb", qfi_method="lin_comb_full", regularization="ridge"
+        )
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_LiH(self):
+        """Lih test"""
+        inter_dist = 1.6
+        driver1 = PySCFDriver(
+            atom="Li .0 .0 .0; H .0 .0 " + str(inter_dist),
+            unit=UnitsType.ANGSTROM,
+            charge=0,
+            spin=0,
+            basis="sto3g",
+        )
+        transformer = ActiveSpaceTransformer(
+            num_electrons=2,
+            num_molecular_orbitals=3,
+        )
+        problem1 = ElectronicStructureProblem(driver1, [transformer])
+        self.expected = -1.85727503
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        calc = AdaptVQE(self.qubit_converter, solver)
+        res = calc.solve(problem1)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
```

Comment:
This is running far too long for the normal CI. Please mark it with `@slow_test` and add `from qiskit.test import slow_test`